### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend operator ReduceMax-11, 12, 13, 18, 20 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20555]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceMax for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-11),  [opset 12](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-12),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-18),  [opset 20](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-20)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceMax)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-11),  [opset 12](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-12),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-18),  [opset 20](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-20)
1. Operator already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by siddhant-0707** [comment]
.take

**Comment by p-wysocki** [comment]
Hello @siddhant-0707, are you still working on this issue?

I am happy to announce that we have created a channel dedicated to Good First Issues support on our [Intel DevHub Discord server](https://discord.gg/wPuqAujS)! Join it to receive support, engage in discussions, ask questions and talk to OpenVINO developers.

**Comment by siddhant-0707** [comment]
@p-wysocki Yes I've figured out what needs to be done, need a little more time

**Comment by siddhant-0707** [comment]
@gkrivor [ONNX ReduceMax](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#type-constraints-252) supports double type, whereas [OV ReduceMax](https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/reduce_max.cpp) doesn't. What do you suggest we do about this?

**Comment by siddhant-0707** [comment]
| Version | Changes |
|---------|---------|
| ReduceMax-1 | Initial version. |
| ReduceMax-11 | Added range for `axes` attribute: `[-r, r-1]` where `r = rank(data)`. |
| ReduceMax-12 | Added support for `tensor(uint8)` and `tensor(int8)` types. |
| ReduceMax-13 | Inputs and outputs are now differentiable. Added support for `tensor(bfloat16)` type. |
| ReduceMax-18 | Added `noop_with_empty_axes` attribute. `axes` input is now optional and non-differentiable. |
| ReduceMax-20 | Added support for `tensor(bool)` type. |

**Comment by p-wysocki** [comment]
Hi @siddhant-0707, can we move forward with this task or is a discussion required?

**Comment by siddhant-0707** [comment]
@p-wysocki I'm working on the task, but a discussion will be required later

**Comment by gkrivor** [comment]
Let me do this as an example
